### DISCUSSION TITLE
Do not deprecate Transaction#hash as it offers support for deserializing websocket responses

### DISCUSSION THREAD

**Discussion by theotherian** [XRPLF/xrpl4j#308] [General]
Currently, the `hash` method on `Transaction` is marked as deprecated, both in `main` and the v3 preview:

https://github.com/XRPLF/xrpl4j/blob/4255c50695077bef51df1c46f20e8f1844adf010/xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/transactions/Transaction.java#L275

https://github.com/XRPLF/xrpl4j/blob/df/v3-dev-preview/xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/transactions/Transaction.java#L267

The reason given is:
> The transaction hash of this transaction.  Only present in responses to `{@code account_tx}` rippled calls.

This isn't true though. When subscribing to a transaction stream via websocket, the only place to find the transaction hash is on the transaction:
https://xrpl.org/subscribe.html#transaction-streams

If someone wants to use xrpl4j to help deserialize results from a websocket stream, losing this field would make it more painful to uniquely identify in upstream eventing. 

By default, unless you ask for "proposed" transactions as well, all transactions should be finalized on the ledger.

Let me know if there's another or better way to identify the transaction.

**Comment by sappenin** [comment]
Interesting, and nice catch @theotherian! 

@nkramer44 -- do you remember why we moved the `hash` value outside of `Transaction`? The new value inside of `AccountTransactionsTransaction` is non-optional, so did we maybe do this so it could be backward-compatible?
